src/main/java/carpet/mixins/UpdateStructureBlockC2SPacketMixin.java

```diff
@@ -0,0 +1,50 @@
+package carpet.mixins;
+
+import carpet.CarpetSettings;
+import net.minecraft.network.PacketByteBuf;
+import net.minecraft.network.packet.c2s.play.UpdateStructureBlockC2SPacket;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(UpdateStructureBlockC2SPacket.class)
+public class UpdateStructureBlockC2SPacketMixin {
+    @Shadow
+    private BlockPos offset;
+    @Shadow
+    private BlockPos size;
+
+    @Inject(
+            method = "read",
+            at = @At("TAIL")
+    )
+    private void structureBlockLimitsRead(PacketByteBuf buf, CallbackInfo ci) {
+        BlockPos previousOffset = offset;
+        BlockPos previousSize = size;
+        try {
+            // This will throw an exception if carpet is not installed on client
```

Comment:
> Also Carpet changes should always be behind a default rule value check, in this case both when sending and receiving.

I thought using `try` is easier than using some buf functions to check that. Will change it later.  
I know that but I didn't want to make another carpet setting to allow setting it to more than 127 (byte limit). I could just do that only if `structureBlockLimit` is more than 127, but when user tries to set it to high value, it will bring it back to 0 instead of the limit value.